### DISCUSSION TITLE
NuGet Replacements for .NET MAUI

### DISCUSSION THREAD

**Discussion by davidortinau** [dotnet/maui#14459] [Upgrading from Xamarin]
One of the main reasons your application will not immediately compile after your initial upgrade to .NET MAUI from Xamarin is incompatible NuGets. The solution is to update them to compatible versions, or replace them with compatible solutions. 

> **Compatible** - to identify compatible frameworks, become familiar with the Frameworks tab when look at packages on [NuGet.org](https://www.nuget.org)

There are three main scenarios you will encounter:

1. NuGets without platform dependencies 
2. NuGets with incompatible platform dependencies and no direct upgrade path
3. NuGets that support both unsupported and supported platform dependencies

NuGets without any platform dependencies (going all the way down through their dependency tree) that have framework support for the "products" .NET Standard or .NET are compatible with .NET MAUI. Some examples are [Newtonsoft.Json](https://www.nuget.org/packages/Newtonsoft.Json), [NLog](https://www.nuget.org/packages/NLog), [Polly](https://www.nuget.org/packages/Polly), and [Refit](https://www.nuget.org/packages/Refit). Note while they may have added .NET target frameworks, there are no dependencies in the Xamarin.*, MonoAndroid, MonoTouch, etc. target frameworks.

For scenarios 2 and 3, so long as you see target framework support for `net6.0-*` or newer then the package is compatible with .NET MAUI.

| Xamarin | .NET MAUI |
|---|---|
| FFImageLoading  | [FFImageLoading.Compat](https://github.com/Redth/FFImageLoading.Compat) or [FFImageLoading.Maui](https://github.com/microspaze/FFImageLoading.Maui)  |
| ZXing.Net.Mobile | [ZXing.Net.Maui](https://github.com/Redth/ZXing.Net.Maui) |
| Xamarin.Controls.SignaturePad | [Drastic.SignaturePad.Maui](https://www.nuget.org/packages/Drastic.SignaturePad.Maui) |
| Telerik UI | https://www.telerik.com/maui-ui  |
| App Center SDK (Crashes & Analytics) | https://www.nuget.org/packages/Microsoft.AppCenter/5.0.0-preview.1 |
| LiveCharts | https://github.com/beto-rodriguez/LiveCharts2  |
| IeuanWalker.StateButton | [IeuanWalker.Maui.StateButton](https://www.nuget.org/packages/IeuanWalker.Maui.StateButton) |
| Xamarin.Forms.Breadcrumb | [IeuanWalkerMaui.Breadcrumb](https://www.nuget.org/packages/IeuanWalker.Maui.Breadcrumb) |
| Xamarin.Forms Plainer | https://github.com/enisn/Xamarin.Forms.Plainer |
| PinchZoom Image | https://github.com/TBertuzzi/Bertuzzi.MAUI.PinchZoomImage |
| Plugin.NFC | https://github.com/franckbour/Plugin.NFC |
| Syncfusion | https://www.syncfusion.com/maui-controls |
| newtonsoft | https://www.nuget.org/packages/Newtonsoft.Json/13.0.2-beta2 |
| Prism | https://www.nuget.org/packages/Prism.Maui/ |
| FresmMVVM | https://www.nuget.org/packages/FreshMvvm.Maui/ |
| Xamarin Community toolkit | [.NET MAUI Community Toolkit](https://docs.microsoft.com/en-us/dotnet/communitytoolkit/maui/get-started) |
| Plugin.Fingerprint | https://www.nuget.org/packages/Plugin.Fingerprint/3.0.0-beta.1 |
| ShinyLib | https://shinylib.net/ |
| Google Materials Lib | https://github.com/albi005/MaterialColorUtilities |
| Serilog | https://github.com/serilog/serilog-sinks-xamarin |
| XCalendar | https://github.com/ME-MarvinE/XCalendar |
| MSAL | https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/tree/main/tests/devapps/MauiApps |
| ReactorUI XF | https://github.com/adospace/reactorui-maui |
| OnScreenSizeMarkup | https://github.com/carolzbnbr/OnScreenSizeMarkup |
| Xamarin Plugin Image Edit | https://github.com/busec0/Xamarin.Plugin.ImageEdit |
| TinyMvvm | https://github.com/TinyStuff/TinyMvvm |
| Xamarin Bluetooth LE | https://github.com/busec0/xamarin-bluetooth-le |
| Sharp Constraint Layout | https://github.com/xtuzy/SharpConstraintLayout |
| Compiled Binding | https://github.com/levitali/CompiledBindings |
| Sentry | https://github.com/getsentry/sentry-dotnet |
| ESRI | https://developers.arcgis.com/net/ Nuget : https://www.nuget.org/packages/Esri.ArcGISRuntime.Maui/100.14.1-preview3 |
| Plugin Local Notification | https://github.com/thudugala/Plugin.LocalNotification |
| Xamanimations | [AlohaKit.Animations](https://github.com/jsuarezruiz/AlohaKit.Animations) |
| Sharpnado Tabs and Presentation | [Sharpnado.Tabs](https://github.com/roubachof/Sharpnado.Tabs) |

**Comment by Kapusch** [comment]
Wow, this is great! Thank you for sharing this list.
May I suggest @davidortinau to mention about the [Xamarin.CommunityToolkit.MauiCompat](https://www.nuget.org/packages/Xamarin.CommunityToolkit.MauiCompat#versions-body-tab) in case people cannot/don't want to migrate too fast ?

**Comment by nll** [comment]
This is a very useful list, thank you for this !

The particular case of _FFImageLoading.Compat_, the nuget publish did fail because `The package ID is reserved`, do you know if that is something that will be eventually fix or it would be better to publish the nuget packages locally ?

**Comment by snakefoot** [comment]
@davidortinau Curious what is needed for NLog to be removed from unsupported-list for MAUI-applications?

Notice NLog v5.2.2 no longer uses assembly-reflection by default, and supports dotnet publish [without trim warnings.](https://nlog-project.org/2023/05/30/nlog-5-2-trim-warnings.html)

See also: https://github.com/NLog/NLog.Targets.MauiLog

**Comment by 007vel** [comment]
Can you suggest replacements for PCL Storage in .Net MAUI?

**Comment by Aguirrehub25** [comment]
Talvez saben si aun se puede usar sidemenuview en .net maui ?
